### PR TITLE
Add an environment variable for disabling the instance group controller

### PR DESCRIPTION
**Pull request by ingwonsong**
This PR proposes a new environment variable `EXPERIMENTAL_ONLY_FOR_TESTING_DISABLE_INSTANCE_GROUP_CONTROLLER` for disabling the instance group controller for the testing env.

When we are doing a test or dev on non-GCP, the instance group controller is not needed, so this env var provides the way to turn off the instance group controller. Please note that it is disabled only when the env var has the value `true` (case insensitive).

### CODE REVIEW

**Comment by aojea** [comment]
/assign @cezarygerard @swetharepakula 

> When we are doing a test or dev on non-GCP, the instance group controller is not needed, so this flag provides the way to turn off the instance group controller. Default is true to keep the compatibility.

exposing a flag just for testing sounds like a footgun for users, can we make this option more hidden?

**Comment by ingwonsong** [comment]
> exposing a flag just for testing sounds like a footgun for users, can we make this option more hidden?

Make sense. Then, what do you think about an env var?

**Comment by aojea** [comment]
> Make sense. Then, what do you think about an env var?

I'm in favor of that,  exaggerating :) `EXPERIMENTAL_ONLY_FOR_TESTING_DISABLE_INSTANCE_GROUP_CONTROLLER`

**Comment by ingwonsong** [comment]
> I'm in favor of that, exaggerating :) `EXPERIMENTAL_ONLY_FOR_TESTING_DISABLE_INSTANCE_GROUP_CONTROLLER`

Sounds great. Will change to use it.

**Comment by ingwonsong** [comment]
> > I'm in favor of that, exaggerating :) `EXPERIMENTAL_ONLY_FOR_TESTING_DISABLE_INSTANCE_GROUP_CONTROLLER`
> 
> Sounds great. Will change to use it.

Done.

File: cmd/glbc/main.go [hunk 1]
```diff
@@ -358,14 +359,16 @@ func runControllers(ctx *ingctx.ControllerContext) {
 
 	ctx.Start(stopCh)
 
-	igControllerParams := &instancegroups.ControllerConfig{
-		NodeInformer: ctx.NodeInformer,
-		IGManager:    ctx.InstancePool,
-		HasSynced:    ctx.HasSynced,
-		StopCh:       stopCh,
+	if strings.ToLower(os.Getenv("EXPERIMENTAL_ONLY_FOR_TESTING_DISABLE_INSTANCE_GROUP_CONTROLLER")) != "true" {
```
Comment (aojea) [inline cmd/glbc/main.go:362]: I suggested this name as a kind of joke but this works for me 😄

Comment (ingwonsong) [reply cmd/glbc/main.go:362]: Yes. it works 😄

**Comment by ingwonsong** [comment]
At this time, just forked version would be enough, because this is just for testing.

**Pull request closed without merging**